tests/components/zwave/test_init.py

```diff
@@ -1310,699 +1235,851 @@ def mock_connect(receiver, signal, *args, **kwargs):
+    zwave_network.nodes = {14: node}
+
+    await hass.services.async_call(
+        "zwave",
+        "change_association",
+        {
+            const.ATTR_ASSOCIATION: "add",
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_TARGET_NODE_ID: 24,
+            const.ATTR_GROUP: 3,
+            const.ATTR_INSTANCE: 5,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert ZWaveGroup.called
+    assert len(ZWaveGroup.mock_calls) == 2
+    assert ZWaveGroup.mock_calls[0][1][0] == 3
+    assert ZWaveGroup.mock_calls[0][1][2] == 14
+    assert group.add_association.called
+    assert len(group.add_association.mock_calls) == 1
+    assert group.add_association.mock_calls[0][1][0] == 24
+    assert group.add_association.mock_calls[0][1][1] == 5
+
+
+async def test_remove_association(hass, mock_openzwave, zwave_setup):
+    """Test zwave change_association service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    await hass.async_block_till_done()
+
+    ZWaveGroup = mock_openzwave.group.ZWaveGroup
+    group = MagicMock()
+    ZWaveGroup.return_value = group
+
+    value = MockValue(index=12, command_class=const.COMMAND_CLASS_WAKE_UP)
+    node = MockNode(node_id=14)
+    node.values = {12: value}
+    node.get_values.return_value = node.values
+    zwave_network.nodes = {14: node}
+
+    await hass.services.async_call(
+        "zwave",
+        "change_association",
+        {
+            const.ATTR_ASSOCIATION: "remove",
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_TARGET_NODE_ID: 24,
+            const.ATTR_GROUP: 3,
+            const.ATTR_INSTANCE: 5,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert ZWaveGroup.called
+    assert len(ZWaveGroup.mock_calls) == 2
+    assert ZWaveGroup.mock_calls[0][1][0] == 3
+    assert ZWaveGroup.mock_calls[0][1][2] == 14
+    assert group.remove_association.called
+    assert len(group.remove_association.mock_calls) == 1
+    assert group.remove_association.mock_calls[0][1][0] == 24
+    assert group.remove_association.mock_calls[0][1][1] == 5
+
+
+async def test_refresh_entity(hass, mock_openzwave, zwave_setup):
+    """Test zwave refresh_entity service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    await hass.async_block_till_done()
+
+    node = MockNode()
+    value = MockValue(
+        data=False, node=node, command_class=const.COMMAND_CLASS_SENSOR_BINARY
+    )
+    power_value = MockValue(data=50, node=node, command_class=const.COMMAND_CLASS_METER)
+    values = MockEntityValues(primary=value, power=power_value)
+    device = get_device(node=node, values=values, node_config={})
+    device.hass = hass
+    device.entity_id = "binary_sensor.mock_entity_id"
+    await hass.async_add_job(device.async_added_to_hass())
```

Comment:
Can we await this directly now instead of wrapping it with `async_add_job`?

Reply:
Yes 👍